Use array to map column to rules

devNoiseConsulting opened the pull request:
- Mapped the column types to validation rules.
- Changed the code to lookup the validation rule instead of the if else code.
- If validation rule is for a string then the max rule is applied as well.

devNoiseConsulting committed:
Clean up

Adding packages so the autocomplete will work.
Adding Duster and Pint to apply code style.
Added .gitignore

devNoiseConsulting committed:
Moving to lookup rules

Scraped the Doctrine/DBAL package for column types. Mapped the column types to validation rules. Changed the code to lookup the validation rule instead of the if else code. If validation rule is for a string then the max rule is applied as well.

ahmetsabri commented on src/Console/ValidateTableCommand.php:
```diff
@@ -21,6 +21,94 @@ class ValidateTableCommand extends Command
+     *
+     * @var array
+     */
+    protected $columnTypeRules = [
```
I liked it thank you, but this array seems too long, could you move it to a json file and resolve it by json_decode() method?
then I will merge it.

devNoiseConsulting replied:
Moved array data to a json file. Can also be published to the app's resources dir for customization.

ahmetsabri replied:
I was about to merge it but I made some changes to to .gitignore file  that caused conflicts to your PR, can you please resolve it :)

devNoiseConsulting replied:
I've fixed the merge conflict. Should be good to merge now.

devNoiseConsulting committed:
Moving to config file

Per feedback moving the rules array to json file to be read in.
Also moved the exclude array to the config file as well.

devNoiseConsulting committed:
User Customization

Added the option to publish the config file into the application's resources.

devNoiseConsulting committed:
Fix code warnings

Change the package dependencies to just be the "laravel/framework". The use of the "resource_path" function gave warnings about the function being undefined.

Also fixed a typo in the readme file.

devNoiseConsulting committed:
Delete installed.php

rebase and merge put kept this file. Removing it again so that the vendor directory is not distributed.

devNoiseConsulting committed:
Change inet to use IP rule

The Postgres column type inet can be validated with the ip rule instead of string.

Also added a markdown file that show column types and database support for them.
